File: string/第一个出现两次的字符.cpp

#include<bits/stdc++.h>

using namespace std;

class Solution {
public:
    char repeatedCharacter(string s) {
        set<char> set2;

        for(auto i:s) {
            if(set2.find(i)==set2.end()) {
                set2.insert(i);
            }
            else {
                return i;
            }
        }

        return 0;
    }
};

int main() {

    string s = "abccbaacz";
    Solution s2;
    cout<<s2.repeatedCharacter(s)<<endl;
    return 0;
}

File: string/第一个出现两次的字符.py



class Solution:
    def repeatedCharacter(self, s: str) -> str:
        occ = set()
        for c in s:
            if c not in occ:
                occ.add(c)
            else:
                return c
        return ''


if __name__ == "__main__":
    str2 = "abccbaacz"
    s2 = Solution()
    print(s2.repeatedCharacter(str2))